green_cli/common.py

```diff
@@ -532,6 +532,35 @@ def mnemonic(file_, mnemonic):
         mnemonic = fileinput.input(mnemonic).readline()
     DefaultAuthenticator(context.options).set_mnemonic(mnemonic)
 
+@green.command()
+@click.option('--method', default="GET", expose_value=False, callback=details_json)
+@click.option('--urls',  default="",multiple=True, expose_value=False, callback=details_json)
+@click.option('--accept',  default="json", type=str, expose_value=False, callback=details_json)
+@click.option('--headers',  type=str, expose_value=False, callback=details_json)
+@click.option('--timeout',  default=10, type=int, expose_value=False, callback=details_json)
+@with_login
+@print_result
+def httprequest(session, details):
+    """Make a request to an http server.
+
+    Request headers may be passed using a python dictionary string, e.g.
+    --headers="{'If-Modified-Since':'Mon, 02 Sep 2019 22:39:39 GMT'}"
```

Comment:
This syntax feels slightly inconsistent as there's nowhere else in the code that uses the same style. There are other examples of passing key/value pairs for example the --addressee of `createtransaction` takes an address and an amount, so you could instead have something like

`--header If-Modified-Since "Mon, 02 Sep 2019 22:39:39 GMT"

maybe

Reply:
moved to a separate PR